server/text_generation_server/layers/fp8.py

```diff
@@ -20,12 +26,20 @@ def get_fp8_linear() -> torch.nn.Module:
     return Fp8Linear
 
 
-def fp8_quantize(weight, qdtype=torch.float8_e4m3fn):
-    device = weight.device
+def fp8_quantize(weight, scale_upper_bound=None, qdtype=torch.float8_e4m3fn):
+    if HAS_FBGEMM:
+        if scale_upper_bound.device != weight.device:
+            scale_upper_bound = scale_upper_bound.to(weight.device)
+
+        qweight, scale = torch.ops.fbgemm.quantize_fp8_per_row(
```

Comment:
Does the quantization kernel work on H100? I guess it does, but if it doesn't, we also have to make this on the CUDA capability.

Reply:
Added check on compute capabilities.